### DISCUSSION TITLE
Metadata upload in Files-Service

### DISCUSSION THREAD

**Discussion by lucascanna** [mia-platform/community#6] [Q&A] [answered]
Hi everyone,

I am trying to upload additional metadata in the files-service as described [here](https://docs.mia-platform.eu/docs/runtime_suite/files-service/usage#upload-post-).

However, once I upload the file along with the metadata, the metadata is not saved in the MongoDB collection.

This is the cURL I am trying to execute:

```
curl -v --request POST \
  --url https://my-project.example.com/files/ \    
  --header 'Authorization: ***' \
  --header 'Content-Type: multipart/form-data' \
  --form "groups=abc;type=text/plain" \
  --form api.png=@/Users/lucascannapieco/Downloads/api.png
```

The server respond with 200 status code and the default metadata in the response body.
However, I would expect to find a `groups` key in the files metadata on the MongoDB collection but this field does not appear.
I have also added the `groups` field in the CRUD schema of the collection.

Do you have any idea of why this is happening?

Thanks in advance

**Comment by davidebianchi** [comment, marked as answer]
Hi @Lucascanna! Have you added a caster file? If yes, can you share it?

As written in the docs: 

```
During the upload, you can specify other properties but they will be ignored. If you like, using the environment variable ADDITIONAL_FUNCTION_CASTER_FILE_PATH you can specify a caster function to add other properties to CRUD.
```

[Here](https://docs.mia-platform.eu/docs/runtime_suite/files-service/configuration#caster-file) should be documented how to write a caster file.

**Comment by lucascanna** [comment]
Yes! I was missing the caster file. Now it works!

Thanks a lot!
